mirascope/core/base/tool.py

```diff
@@ -85,6 +85,9 @@ def call(self) -> str:
     tool_config: ClassVar[ToolConfig] = ToolConfig()
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
+    if TYPE_CHECKING:
+        tool_call: Any
```

Comment:
what is the benefit of including this? if the answer is so that a user doesn't accidentally add a `tool_call` field to their `BaseTool` definition, then we should remove `if TYPE_CHECKING` and just always have `tool_call` as a field of `BaseTool`, right?

Separately, it seems adding this results in a bunch of pyright ignores (e.g. in our tests). I'm fine with that if we think this is worthwhile, but would like to see if there's a better solution (e.g. making `tool_call` optional for `BaseTool` but required for provider-specific tools e.g. `OpenAITool`?)

Last, if we are going to include this, we should ensure it is skipped for json schema generation like we do for tool calls for all provider-specific tool types

Reply:
I add the filed in BaseTool

Reply:
I tried adding a Field in `BaseTool`, but it does not seem possible to set a default value for a Field in `BaseTool` and not set the default in the inheriting class. (pyright error)
It seems to be referencing tool_call without checking it when calling it in the first place, so if all inherited classes don't have `tool_call`, it should be checked